RevenueCatUI/Helpers/StoreProductDiscount+Extensions.swift

```diff
@@ -12,9 +12,30 @@
 //  Created by Cesar de la Vega on 25/9/24.
 
 import Foundation
-import RevenueCat
+@_spi(Internal) import RevenueCat
 
-extension StoreProductDiscount {
+@_spi(Internal) extension StoreProductDiscountType {
+
+    var discountedPeriodsWithUnit: String {
```

Comment:
Why can't this property keep being part of the `StoreProductDiscount` and has to be defined in the `StoreProductDiscountType` extension? Perhaps I'm missing something (or maybe the why will come in a future PR), but it looks to me that it could remain as a variable in `StoreProductDiscount`

Reply:
Let me clarify.

This is indeed needed for an upcoming PR. We'll include `StoreProductDiscountType` in `PurchaseInformation` so we can work with discounts and display the most accurate billing information.

To be able to work with mocks I need the protocol, and all the available methods to use. In this case, the one I need is `localizedPricePerPeriodByPaymentMode`